tests/test_google_ads_interrupted_sync_full_table.py

```diff
@@ -0,0 +1,121 @@
+
+from base import GoogleAdsBase
+
+
+class InterruptedSyncFullTableTest(GoogleAdsBase):
+    """Test tap's ability to recover from an interrupted sync for FULL Table stream"""
+
+    @staticmethod
+    def name():
+        return "tt_google_ads_interruption_full_table"
+
+    def test_run(self):
+
+        # call_details, campaign_labels do not have an id field
+        # accessible_bidding_strategies, accounts, user_list, and bidding_strategies streams have only 1 record.
+        # So, skipped those streams.
+        streams_to_test = {'ad_groups': 132093547633, 'ads': 586722860308, 'campaign_budgets': 10006446850,
+                           'campaigns': 15481829265, 'labels': 21628120997, 'carrier_constant': 70094, 'feed': 351805305,
+                           'feed_item': 216977537909, 'language_constant': 1007, 'mobile_app_category_constant': 60009,
+                           'mobile_device_constant': 604043, 'operating_system_version_constant': 630166, 'topic_constant': 41,
+                           'user_interest': 959, 'campaign_criterion': 16990616126, 'ad_group_criterion': 131901833709}
+
+        for stream, last_pk_fetched in streams_to_test.items():
+            self.run_test(stream, last_pk_fetched)
+
+    def run_test(self, stream, last_pk_fetched):
+        """
+        Scenario: A sync job is interrupted for full table stream. The state is saved with `currently_syncing`
+                    and `last_pk_fetched`(id of last synced record).
+                  The next sync job kicks off, the tap picks only remaining records for interrupted stream and complete the sync.
+
+        Test Cases:
+            - Verify id value of each record in interrupted sync is less than last_pk_fetched(id of last synced record).
+            - Verify that all records in the full sync and interrupted sync come in Ascending order.
+            - Verify interrupted_sync has the fewer records as compared to full sync
+            - Verify state is flushed if sync is completed.
+        """
+
+        # Create connection using a recent start date
+        conn_id = connections.ensure_connection(self)
+
+        # Run a discovery job
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # partition catalogs for use in table/field seelction
+        test_catalogs = [catalog for catalog in found_catalogs
+                           if catalog.get('stream_name') in {stream}]
+
+        # select fields
+        self.select_all_streams_and_fields(conn_id, test_catalogs, select_all_fields=False)
+
+        # Run a sync
+        self.run_and_verify_sync(conn_id)
+
+        # acquire records from target output
+        full_sync_records = runner.get_records_from_target_output()
+
+        interrupted_state = {
+            'currently_syncing': (stream, '5548074409'),
+            'bookmarks': {
+                stream: {'5548074409': {'last_pk_fetched': last_pk_fetched}},
+           }
+        }
```

Comment:
Instead of testing every stream individually, I think a more valuable approach is to validate that the sync can resume when it is interrupted part-way through with multiple streams selected. That way we can verify the ordering of streams is maintained on the resuming sync. So in a sync replicating streams A, B and C the interrupted state would be set to stream B. Then the resuming sync should start with stream B, and then move on to stream C. At the tap-tester level we can validate the expected records are synced by comparing to the uninterrupted sync results. At the unittest level the ordering should be explicitly tested. I'd expect the streams to be ordered alphabetically by name, and that the order is shuffled to account for interrupted states.